client/js/jsonedit.js

```diff
@@ -278,6 +277,19 @@ const jeCommands = [
       jeSetAndSelect(null);
     }
   },
+  {
+    id: 'je_duplicateWidget',
+    name: '✨ duplicate widget',
+    forceKey: 'D',
+    show: _=>jeStateNow,
+    call: function() {
+      var currentWidget = JSON.parse(JSON.stringify(jeWidget.state))
+      currentWidget.id = null;
+      addWidgetLocal(currentWidget);
+      jeClick(widgets.get(currentWidget.id), true);
+      jeSetAndSelect(null);
```

Comment:
Maybe add `jeStateNow.id = '###SELECT ME###'` before the `jeSetAndSelect` so it highlights the new ID?